### ISSUE TITLE
Upgrade version of https-proxy-agent

### ISSUE THREAD

**Issue by mjalkio** [serverless/serverless#4899]
# This is a Bug Report

## Description

I am using a tool called Snyk which checks dependencies for security vulnerabilities. Currently, serverless is considered a vulnerable project due to one of its dependencies, [https-proxy-agent](https://github.com/TooTallNate/node-https-proxy-agent). The vulnerability can be read about [here](https://hackerone.com/reports/319532).

It was fixed in 2.2.0, which at a glance it looks like serverless could upgrade to.

Similar or dependent issues:
* Similar to a previous issue I opened, #4464

## Additional Data

* ***Serverless Framework Version you're using***: N/A
* ***Operating System***: N/A
* ***Stack Trace***: N/A
* ***Provider Error messages***: N/A

**Comment by HyperBrain** [comment]
Hi @mjalkio , good catch 👍

**Comment by hxoht** [comment]
Is this supposed to be resolved in `serverless@1.27.3`?

```
> serverless@1.27.3 postinstall /Users/paolofragomeni/projects/mpg/image-press-serverless/node_modules/serverless
> node ./scripts/postinstall.js

(node:84859) ExperimentalWarning: The fs.promises API is experimental
added 823 packages from 524 contributors in 25.381s
[!] 1 vulnerability found [4935 packages audited]
    Severity: 1 High
    Run `npm audit` for more detail
```

**Comment by hxoht** [comment]
Is this supposed to be resolved in `serverless@1.27.3`?

```
> serverless@1.27.3 postinstall ./node_modules/serverless
> node ./scripts/postinstall.js

(node:84859) ExperimentalWarning: The fs.promises API is experimental
added 823 packages from 524 contributors in 25.381s
[!] 1 vulnerability found [4935 packages audited]
    Severity: 1 High
    Run `npm audit` for more detail
```

**Comment by HyperBrain** [comment]
No, the merged PR is tagged for 1.28, but we can try to publish a 1.27.4 with the dependency updates (there are others waiting too)

**Comment by pritamstyz4ever** [comment]
Hi, I am waiting on the fix for  "https-proxy-agent": "^2.2.0" which will fix the vulnerability, when is the fix for this expected?
